client/asset/btc/btc.go

```diff
@@ -6008,14 +6008,53 @@ func (btc *intermediaryWallet) WalletTransaction(ctx context.Context, txID strin
 	}
 
 	txs, err := btc.TxHistory(1, &txID, false)
-	if err != nil {
+	if err != nil && !errors.Is(err, asset.CoinNotFoundError) {
 		return nil, err
 	}
-	if len(txs) == 0 {
-		return nil, asset.CoinNotFoundError
+
+	var tx *asset.WalletTransaction
+	if len(txs) > 0 {
+		tx = txs[0]
+	}
+	if tx == nil {
+		txHash, err := chainhash.NewHashFromStr(txID)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding txid %s: %w", txID, err)
+		}
+
+		gtr, err := btc.node.getWalletTransaction(txHash)
+		if err != nil {
+			return nil, fmt.Errorf("error getting transaction %s: %w", txID, err)
+		}
+
+		var blockHeight uint32
+		if gtr.BlockHash != "" {
+			blockHash, err := chainhash.NewHashFromStr(gtr.BlockHash)
+			if err != nil {
+				return nil, fmt.Errorf("error decoding block hash %s: %w", gtr.BlockHash, err)
+			}
+			height, err := btc.tipRedeemer.getBlockHeight(blockHash)
+			if err != nil {
+				return nil, fmt.Errorf("error getting block height for %s: %w", blockHash, err)
+			}
+			blockHeight = uint32(height)
+		}
+
+		tx, err = btc.idUnknownTx(&ListTransactionsResult{
+			BlockHeight: blockHeight,
+			BlockTime:   gtr.BlockTime,
+			TxID:        txID,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error identifying transaction: %v", err)
+		}
```

Comment:
Should add this unknown tx to the db? Also for dcr.

Reply:
> Should add this unknown tx to the db? Also for dcr.

I was also hoping to get an answer to this question.

Reply:
Sorry, yes, I've updated it to add it.